src/core/qgsexpressionlexer.ll

```diff
@@ -220,17 +220,17 @@ string      "'"{str_char}*"'"
 
 {string}  { TEXT_FILTER(stripText); return STRING; }
 
-{deprecated_function} { TEXT; return FUNCTION; }
+{deprecated_function} { TEXT; return NAME; }
 
 {named_node} { TEXT_FILTER(stripNamedText); return NAMED_NODE; }
 
 {special_col}        { TEXT; return SPECIAL_COL; }
 
 {variable}        { TEXT; return VARIABLE; }
 
-{column_ref}         { TEXT; return QgsExpression::isFunctionName(*yylval->text) ? FUNCTION : COLUMN_REF; }
+{column_ref}         { TEXT; return NAME; }
 
-{column_ref_quoted}  { TEXT_FILTER(stripColumnRef); return COLUMN_REF; }
+{column_ref_quoted}  { TEXT_FILTER(stripColumnRef); return NAME; }
```

Comment:
An unexpected side feature - this allows us to call functions like this: ```"sin"(1)``` - which is fine I guess, Postgres also allows that :)